util/sha256fixed/sha256.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sha256fixed is like crypto/sha256,
+// but efficiently hashes fixed-width integers.
+package sha256fixed
```

Comment:
sha256x? "Package sha256 is like crypto/sha256 with extra methods. It exports a concrete Hasher
type rather than only returning an interface implementation."

Reply:
(Because I suspect this'll grow more stuff than just a few integer hashing methods)

Reply:
SGTM.